population/__init__.py

```diff
@@ -4,27 +4,25 @@ class Population:
     def __init__(self, popSize, geneLength):
         self.popSize = popSize
         self.geneLength = geneLength
-        self.fittestScore = 0
         self.individuals = []
         for i in range(popSize):
             self.individuals.append(Individual(self.geneLength))
 
-    def selectFittest(self):
-        maxFit = 0
-        maxFitIndex = 0
-        for individual in self.individuals:
-            individual.setFitness(individual.calcFitness())
-            if individual.getFitness() > maxFit:
-                maxFit = individual.getFitness()
-                maxFitIndex = self.individuals.index(individual)
-        self.fittestScore = maxFit
-        try:
-            cloneOfFittest = Individual(self.geneLength)
-            cloneOfFittest.setGenes(self.individuals[maxFitIndex].getGenes())
-            cloneOfFittest.setFitness(self.individuals[maxFitIndex].getFitness())
-            return cloneOfFittest
-        except:
-            print("Could not clone individual in index " + maxFitIndex)
+    @property
+    def fittest(self):
+        fittest, _ = self.selectfittestpair()
+        return fittest
+    @property
+    def secondFittest(self):
+        _, secondFittest = self.selectfittestpair()
```

Comment:
i squashed the camelcasing when I copied this -- should be selectFittestPair

Reply:
thanks, copy-paste errors galore